redis-persistence/src/main/java/com/netflix/conductor/redis/config/RedisClusterConfiguration.java

```diff
@@ -32,9 +34,10 @@ protected JedisCommands createJedisCommands(RedisProperties properties, Conducto
         HostSupplier hostSupplier, TokenMapSupplier tokenMapSupplier) {
         GenericObjectPoolConfig genericObjectPoolConfig = new GenericObjectPoolConfig();
         genericObjectPoolConfig.setMaxTotal(properties.getMaxConnectionsPerHost());
-        Host host = hostSupplier.getHosts().get(0);
-        return new JedisCluster(
-            new redis.clients.jedis.JedisCluster(new HostAndPort(host.getHostName(), host.getPort()),
-                genericObjectPoolConfig));
+        Set hosts = new HashSet<>();
```

Comment:
This could be simplified using streams API:
```java
hostSupplier.getHosts().stream()
            .map(h -> new HostAndPort(h.getHostName(), h.getPort()))
            .collect(Collectors.toSet());
```